Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Internal/DriverMode.hs

```diff
@@ -40,5 +46,42 @@ setDriverMode apiKey req = do
   locationTrackingServiceKey <- asks (.locationTrackingServiceKey)
   unless (apiKey == Just locationTrackingServiceKey) $ do
     throwError $ InvalidRequest "Invalid API key"
-  void $ QDriverInformation.updateActivity isActive (Just mode) driverId
+  let newFlowStatus = getDriverFlowStatus (Just mode) isActive
+  updateFleetOperatorStatusKeyForDriver driverId newFlowStatus Nothing
+  void $ QDriverInformation.updateActivity isActive (Just mode) (Just newFlowStatus) driverId
```

Comment:
`updateActivity` and `updateFleetOperatorStatusKeyForDriver` always should be used together. Maybe add some helper function?